src/jit/codegenarm.cpp

```diff
@@ -1565,6 +1784,158 @@ void CodeGen::genLeaInstruction(GenTreeAddrMode* lea)
 }
 
 //------------------------------------------------------------------------
+// genCompareLong: Generate code for comparing two longs when the result of the compare
+// is manifested in a register.
+//
+// Arguments:
+//    treeNode - the compare tree
+//
+// Return Value:
+//    None.
+//
+// Comments:
+// For long compares, we need to compare the high parts of operands first, then the low parts.
+// If the high compare is false, we do not need to compare the low parts. For less than and
+// greater than, if the high compare is true, we can assume the entire compare is true.
+//
+void CodeGen::genCompareLong(GenTreePtr treeNode)
```

Comment:
My plan is to leave these copies of `genCompareLong`, `genJccLongLo` and `genJccLongHi` for now, because they (indirectly) provide more of input context to test this overall `long` thing. I guess it will be quite simple to remove them in future when we have cmp lowering.